lib/index.d.ts

```diff
@@ -56,18 +51,27 @@ export interface ServiceRequest {
     };
 }
 
-export interface ServiceClientResponse extends ResponseObject {
-    readonly payload:
-        | stream.Readable
+export type ServiceContext = {
+    dataSources: {[serviceClient: string]: ClientInstance};
+    request: ServiceRequest;
+};
+
+type ServiceClientResponsePayload = stream.Readable
         | Buffer
         | string
         | {[key: string]: string | string[]};
+
+export interface ServiceClientResponse extends ResponseObject {
```

Comment:
I think doing this would eliminate the need for the `T extends ServiceClientResponsePayload...` line further below...
```suggestion
export interface ServiceClientResponse extends ReturnType<typeof Wreck.request> {
```

Reply:
Nah, I will just keep `ServiceClientResponse` as it is for backward-compatibility

Reply:
I'm not sure it ever should have been ResponseObject in the first place (since this library may not necessarily even be used in conjunction with Hapi). But you're right that it could be breaking.